docs/notebooks/util/plotting.py

```diff
@@ -208,6 +209,101 @@ def plot_bo_points(
             ax.scatter(pts[i, 0], pts[i, 1], obs_values[i], c=col_pts[i], marker=mark_pts[i])
 
 
+def plot_bo_points_in_obj_space(
+    obs_values,
+    num_init=None,
+    mask_fail=None,
+    figsize=None,
+    xlabel="Obj 1",
+    ylabel="Obj 2",
+    zlabel="Obj 3",
+    title=None,
+    m_init="x",
+    m_add="o",
+    c_pass="tab:green",
+    c_fail="tab:red",
+    c_pareto="tab:purple",
+    only_plot_pareto=False,
+):
+    """
+    Adds scatter points in objective space, used for multi-objective optimization (2 objective only).
+    Markers and colors are chosen according to BO factors.
+    :param obs_values:
+    :param num_init: initial number of BO points
+    :param mask_fail: Boolean vector, True if the corresponding observation violates the constraint(s)
+    :param title:
+    :param xlabel:
+    :param ylabel:
+    :param figsize:
+    :param only_plot_pareto: if set true, only plot the pareto points
+    """
+    obj_num = obs_values.shape[-1]
+    tf.debugging.assert_shapes([])
+    assert obj_num == 2 or obj_num == 3, NotImplementedError(
```

Comment:
I thought it only works with 2?

Reply:
For 3 outcomes the plot are expected to show something like [this](https://pymoo.org/_images/problems_many_dtlz_10_0.png).